File: python/2244. Minimum Rounds to Complete All Tasks/solution.py
class Solution:
    def minimumRounds(self, tasks: List[int]) -> int:
        freq = {}

        for difficulty in tasks:
            freq[difficulty] = freq.get(difficulty, 0) + 1

        count = 0
        for diff, occurrence in freq.items():
            if occurrence == 1:
                return -1
            elif occurrence % 3 == 0:
                count += occurrence // 3
            else:
                count += occurrence // 3 + 1

        return count

File: java/2244. Minimum Rounds to Complete All Tasks/Solution.java
class Solution {
    public int minimumRounds(int[] tasks) {
        Map<Integer, Integer> freq = new HashMap<>();
        for (int diff : tasks) freq.put(diff, freq.getOrDefault(diff, 0) + 1);

        int count = 0;
        for (int diff : freq.keySet()) {
            int occurrence = freq.get(diff);
            if (occurrence == 1) return -1;
            else if (occurrence % 3 == 0) count += occurrence / 3;
            else count += occurrence / 3 + 1;
        }
        return count;
    }
}